pkg/types/types.go

```diff
@@ -1,7 +1,25 @@
 package types
 
+import "fmt"
+
 type Library struct {
+	Id      string
 	Name    string
 	Version string
 	License string `json:",omitempty"`
 }
+
+type Dependency struct {
+	Id        string
+	DependsOn []string
+}
+
+func NewLibrary(name, version, license string) Library {
+	// TODO replace naive implementation
+	return Library{
+		Id:      fmt.Sprintf("%s@%s", name, version),
+		Name:    name,
+		Version: version,
+		License: license,
+	}
+}
```

Comment:
What if we just define `ID()`?

```
func (lib Library) ID() string {
	return fmt.Sprintf("%s@%s", name, version)
}
```

Reply:
This is good question. Let's keep it as option. Once everything else is implemented we can return back and review it again

Reply:
It forces to add `types.NewLibrary()` everywhere and it makes review hard. It would be fine if it makes sense, but this ID is kind of like another format of `String()` because `Library` holds `name` and `version`. It looks like just one of expressions of `Library`. But yes, you can change it later.